sklearn/covariance/tests/test_graph_lasso.py

```diff
@@ -26,21 +28,24 @@ def test_graph_lasso(random_state=0):
     X = random_state.multivariate_normal(np.zeros(dim), cov, size=n_samples)
     emp_cov = empirical_covariance(X)
 
-    for alpha in (0., .01, .1):
+    for alpha in (0., .1, .25):
         covs = dict()
+        icovs = dict()
         for method in ('cd', 'lars'):
-            cov_, _, costs = graph_lasso(emp_cov, alpha=alpha,
-                                         return_costs=True)
+            cov_, icov_, costs = graph_lasso(emp_cov, alpha=alpha, mode=method,
+                                             return_costs=True, tol=tol)
             covs[method] = cov_
+            icovs[method] = icov_
             costs, dual_gap = np.array(costs).T
             # Check that the costs always decrease (doesn't hold if alpha == 0)
             if not alpha == 0:
-                assert_array_less(np.diff(costs), 0)
+                assert_array_less(np.diff(costs), 1e-3)
```

Comment:
For badly-scaled problems the change in cost can be slightly positive, on the order of say 1e-12...that was causing failures for some test cases I'd tried, but now I guess it would be fine to make it 0, or maybe 1000*machine epsilon?